mayastor/src/lvs/lvol.rs

```diff
@@ -330,4 +352,55 @@ impl Lvol {
             }
         }
     }
+
+    /// Format snapshot name
+    /// base_name is the nexus or replica UUID
+    pub fn format_snapshot_name(base_name: &str, snapshot_time: u64) -> String {
+        format!("{}-snap-{}", base_name, snapshot_time)
+    }
+
+    /// Create a snapshot
+    pub async fn create_snapshot(
+        &self,
+        nvmf_req: &NvmfReq,
+        snapshot_name: &str,
+    ) {
+        extern "C" fn snapshot_done_cb(
+            nvmf_req_ptr: *mut c_void,
+            _lvol_ptr: *mut spdk_lvol,
+            errno: i32,
+        ) {
+            let nvmf_req = NvmfReq(
+                NonNull::new(nvmf_req_ptr as *mut spdk_nvmf_request).unwrap(),
+            );
+            let mut rsp = nvmf_req.response();
+            let nvme_status = rsp.status();
+
+            nvme_status.set_sct(0); // SPDK_NVME_SCT_GENERIC
+            nvme_status.set_sc(match errno {
+                0 => 0,
+                _ => {
+                    debug!("vbdev_lvol_create_snapshot errno {}", errno);
+                    0x06 // SPDK_NVME_SC_INTERNAL_DEVICE_ERROR
+                }
+            });
+
+            // From nvmf_bdev_ctrlr_complete_cmd
+            unsafe {
+                spdk_sys::spdk_nvmf_request_complete(nvmf_req.0.as_ptr());
+            }
+        }
+
+        let c_snapshot_name = snapshot_name.into_cstring();
+        unsafe {
+            vbdev_lvol_create_snapshot(
+                self.0.as_ptr(),
+                c_snapshot_name.as_ptr(),
+                Some(snapshot_done_cb),
+                nvmf_req.0.as_ptr().cast::<c_void>(),
```

Comment:
Is the turbofish needed? would .cast() not infer void from the function itself?

Reply:
Indeed it isn't, spot the C++ coder.